code/modules/mob/living/simple_animal/abnormality/zayin/pile_of_mail.dm

```diff
@@ -293,3 +326,152 @@
 	if (!TMPEFF)
 		user.apply_status_effect(/datum/status_effect/workspeed_buff)
 	qdel(src)
+
+/obj/item/mailpaper/trapped
+	name = "From the Abnormalities"
+	desc = "Does something automatically or when opened."
+	var/datum/timedevent/effect_timer
+	var/effect_min_time = 6 SECONDS
+	var/effect_max_time = 12 SECONDS
+
+/obj/item/mailpaper/trapped/Initialize()
+	. = ..()
+	effect_timer = new(CALLBACK(src, .proc/Trap), rand(effect_min_time, effect_max_time))
+	playsound(get_turf(src), 'sound/abnormalities/mailpile/gotmail.ogg', 50, 1)
+
+/obj/item/mailpaper/trapped/attack_self(mob/user)
+	to_chat(user, "<span class='warning'>What the-</span>")
+	Trap()
+
+/obj/item/mailpaper/trapped/proc/Trap()
+	set waitfor = FALSE
+	if(effect_timer)
+		deltimer(effect_timer)
+	qdel(src)
+	return
+
+/obj/item/mailpaper/trapped/fairies
+	desc = "Why does this paper smell faintly of Pixy Stix?"
+	var/fairy_count = 4
+	var/weaken_fairy = FALSE
+
+/obj/item/mailpaper/trapped/fairies/Trap()
+	var/turf/T = get_turf(src)
+	T.visible_message("<span class='warning'>[fairy_count > 1 ? "Ravenous fairies" : "A ravenous fairy"] burst from the mail!</span>")
+	for(var/i = 1 to fairy_count)
+		var/mob/living/simple_animal/hostile/mini_fairy/MF =  new(T)
+		MF.faction += "pink"
+		if(weaken_fairy)
+			MF.adjustBruteLoss(43)
+			break
+	return ..()
+
+/obj/item/mailpaper/trapped/acid
+	desc = "Why does this paper smell faintly of battery acid?"
+
+/obj/item/mailpaper/trapped/acid/Trap()
+	var/turf/T = get_turf(src)
+	T.visible_message("<span class='warning'>Acid sprays from the letter!</span>")
+	for(var/i = 1 to 8)
+		var/angle = rand(0, 360)
+		var/obj/effect/decal/cleanable/wrath_acid/bad/AB = new(get_turf(src))
+		MoveAcidAngle(AB, angle)
+	return ..()
+
+/obj/item/mailpaper/trapped/acid/proc/MoveAcidAngle(obj/effect/decal/cleanable/wrath_acid/bad/bad_acid, angle)
+	set waitfor = FALSE
+	var/turf/target_turf = get_turf_in_angle(angle, get_turf(src), pick(1, 2, 4))
+	while(step_towards(bad_acid, target_turf, 0))
+		stoplag(2)
+	return
+
+/obj/item/mailpaper/trapped/urgent
+	desc = "The paper has been stamped 'Urgent'."
+	effect_min_time = 10 SECONDS
+	effect_max_time = 10 SECONDS
+
+/obj/item/mailpaper/trapped/urgent/attack_self(mob/user)
+	to_chat(user, "<span class='notice'>If don't read this within 10 seconds we're going to kill you.</span>\n<span class='nicegreen'>Well, you read it fast enough so that's nice!</span>")
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		H.adjustSanityLoss(-20)
+	deltimer(effect_timer)
+	qdel(src)
+	return
+
+/obj/item/mailpaper/trapped/urgent/Trap()
+	audible_message("<span class='warning'>We are going to kill you.</span>")
+	for(var/mob/living/carbon/human/H in hearers(7, src))
+		H.apply_damage(50, WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE))
```

Comment:
Dealing 50 white damage to everyone in view might be a bit too strict for some random clerk/agent not opening a letter. Either reduce the damage or radius.
Another thing to point out is that you can just... throw the mail away.

Reply:
Yeah, you can. These are meant to catch people off-guard and otherwise be bombs thrown at you. Not a lot get sent out, maybe 1 per 3 people. They disrupt the initial "we set up for the midnight that just arrived" process